### ISSUE TITLE
popup help interferes with editing/copying/pasting of cells

### ISSUE THREAD

**Issue by yanne** [robotframework/RIDE#1043]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004) by scath... on 5 Jul 2012*

Windows 7 (32bit) and Gentoo Linux (64bit), both with:

Python 2.7.3
wxPython 2.8.12.1

This is the same as in <a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=929" title="popup help interferes with editing/copying/pasting of cells"> issue 929 </a>:

"When you mouse over a cell in RIDE, a popup help window appears.  You have to click the cell and very quickly select it for editing or copying, otherwise the popup reappears and prevents you from doing anything in that cell.  This is happening in RIDE since ver 0.39 to current v0.41.  Maybe having a way to turn off the popup help in settings would help this?"

<a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=929" title="popup help interferes with editing/copying/pasting of cells"> issue 929 </a> was written against MacOSX and is marked as "fixed". However i see exactly the same problem in Linux and Windows (cannot retest with MacOSX). The problem is that the popup will appear under the mouse cursor making it impossible to edit this cell until you move the mouse again (please see attached screenshot). please make this popup configurable or place it beside the cursor.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c1) by jerry57 on 6 Jul 2012*


This is also a problem on Mac OS X and has been reported previously, but I don't have the issue number handy.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c2) by @jussimalinen on 20 Jul 2012*


Version 0.41 is no longer the current. Current latest version is 0.46.1. Can you try with that if the behavior changes?

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c3) by @jussimalinen on 20 Jul 2012*


Actually yes, it does affect the latest version. There really should be no reason why the popup should eat the click...

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c4) by jerry57 on 20 Jul 2012*


Yes this is still a problem on Mac OS X as well.  Not only does it swallow the mouse click, but prevents typing in the cell as well.  User has to be quick once you click on the cell or move mouse to a "greyed cell" or out of the grid in order to type in the cells.  Once typing begins, then there isn't a problem.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c5) by jerry57 on 20 Jul 2012*


Forgot to mention, this was with Ride 0.46.1, Mac OS X 10.6, Python 2.6.1 wxPython 2.8.8.1

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c6) by scath... on 29 Jul 2012*


i fixed it for my local installation by editing src\robotide\editor\tooltips.py

changed:

    def _calculate_tooltip_position(self):
        x, y = wx.GetMousePosition()
        return x - 2, y - 2

to:

    def _calculate_tooltip_position(self):
        x, y = wx.GetMousePosition()
        return x + 16, y

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c7) by @mkorpela on 30 Jul 2012*


The current functionality for Windows and Linux is intentional and it is quite different from the OS X problem ( <a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=929" title="popup help interferes with editing/copying/pasting of cells"> issue 929 </a> ). It has also worked like it is working for a long time.

The basic idea is that the tooltip will be shown until mouse position goes outside of the tooltip window OR the user does a keyboard action (like starts typing which will automatically edit the focused cell). Tooltip will be shown only once the mouse cursor has not moved for 1 second.

Could you explain why it is a problem that you have to move the mouse cursor?

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c8) by scath... on 30 Jul 2012*


If the current selected cell is not the cell under the cursor then the tooltip will disappear when starting typing but you will type into the selected cell.

So if you move the cursor over the cell you want to edit it's likely that the tooltip will appear and you have to move the mouse again.

I known it's only a small issue, but it can be annoying :-) Maybe the tooltip should be configurable so a user can turn it off or increase the time until the tooltip appears? Or press a button like "ctrl" like showing the help tooltip?

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c9) by @mkorpela on 3 Aug 2012*


I see why the current functionality might be somewhat annoying but for now we've decided to WontFix this issue.

The risk I see is that there are more (most likely a lot more) then 1000 active RIDE users and the change might generate much more annoyance ("Why can't I copy the argument name from the popup anymore??").

We will of course reconsider if more similar complaints are reported. Thanks for the report.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c10) by jerry57 on 3 Aug 2012*


I have to agree with the option of making it configurable to pop up or not.  Having a way to turn off the tool tip would allow for those of us that this is annoying to just turn it off.  Please reconsider making this a configurable option!

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c11) by @pekkaklarck on 21 Nov 2012*


Fixing this my moving the pop-up a little as suggested in comment 6 sounds like a good idea. Alternatively showing the pop-up could be made configurable. Let's see could either of these, preferably the former, be done for 1.0.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c12) by @pekkaklarck on 21 Nov 2012*


Ought to be fixed now:
<a href="https://github.com/robotframework/RIDE/commit/be857488a7d0a941b2959f78e5bce26142216b9e" rel="nofollow">https://github.com/robotframework/RIDE/commit/be857488a7d0a941b2959f78e5bce26142216b9e</a>

Decided to move the tooltip 16px left and down from the cursor. This keeps the cell below the cursor fully visible in most cases and is about the same location where Chrome browser opens tooltips.

Still need to test this on Windows before closing the issue.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1004#c13) by @pekkaklarck on 21 Nov 2012*


Looks good also on Windows.
